crates/test-programs/tests/wasm_tests/runtime.rs

```diff
@@ -55,41 +55,14 @@ pub fn instantiate(data: &[u8], bin_name: &str, workspace: Option<&Path>) -> any
         .context("failed to instantiate wasi")?,
     );
 
-    // ... and then do the same as above but for the old snapshot of wasi, since
-    // a few tests still test that
-    let mut builder = wasi_common::old::snapshot_0::WasiCtxBuilder::new()
```

Comment:
Will the changes to this file preclude the ability to have test programs that specifically target `wasi_unstable` or older snapshots?

My concern is that we should likely (in the future) have test programs that are specifically designed to test our back compat.

Perhaps we can leave this as-is and then add support for older WASIs when such test programs are implemented.

Reply:
Excellent point! I'm of the opinion we should have tests for different snapshot versions in the future. However, I'm not sure we should have it the way it is ATM. I'm not saying it's bad, just that perhaps we could come up with a more automated way? That said, I'm happy to revert this bit code of back for posterity and future guidance.

Reply:
Personally, I think we should keep your change and in the future add additional test crates that are designed against specific versions of WASI for ensuring old test programs continue to work.  I think having many different target WASIs mixed into one crate can be pretty confusing.

Reply:
@sunfishcode @alexcrichton your thoughts on this?

Reply:
@peterhuene's suggestion sounds good to me.